src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -176,24 +176,35 @@ private static bool ShouldSoftSelectInArgumentList(CompletionContext completionC
         private static bool IsArgumentListTriggerCharacter(char character)
             => character is ' ' or '(' or '[';
 
-        private static int ComputeSymbolMatchPriority(ISymbol symbol)
+        private static int ComputeSymbolMatchPriority(ISymbol symbol, bool preselect)
         {
-            if (symbol.MatchesKind(SymbolKind.Local, SymbolKind.Parameter, SymbolKind.RangeVariable))
+            if (preselect)
             {
-                return SymbolMatchPriority.PreferLocalOrParameterOrRangeVariable;
-            }
+                if (symbol.MatchesKind(SymbolKind.Local, SymbolKind.Parameter, SymbolKind.RangeVariable))
+                {
+                    return SymbolMatchPriority.PreferLocalOrParameterOrRangeVariable;
+                }
 
-            if (symbol.MatchesKind(SymbolKind.Field, SymbolKind.Property))
-            {
-                return SymbolMatchPriority.PreferFieldOrProperty;
+                if (symbol.MatchesKind(SymbolKind.Field, SymbolKind.Property))
+                {
+                    return SymbolMatchPriority.PreferFieldOrProperty;
+                }
+
+                if (symbol.MatchesKind(SymbolKind.Event, SymbolKind.Method))
+                {
+                    return SymbolMatchPriority.PreferEventOrMethod;
+                }
+
+                return SymbolMatchPriority.PreferType;
             }
 
-            if (symbol.MatchesKind(SymbolKind.Event, SymbolKind.Method))
+            // Give locals a small boost in MatchPriority over other non-preselected symbols, since it's most likely to be used.
+            if (symbol.MatchesKind(SymbolKind.Local, SymbolKind.Parameter, SymbolKind.RangeVariable))
             {
-                return SymbolMatchPriority.PreferEventOrMethod;
+                return SymbolMatchPriority.LocalOrParameterOrRangeVariable;
```

Comment:
This would ensure that locals would be given a priority for selection over other symbols when text matching are equally good, and no MRU involved.